src/app/components/restaurants/RestaurantsMenu.jsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import FlatButton from '../utils/FlatButton.jsx';
+import SearchInput from '../utils/SearchInput.jsx';
+
+class RestaurantsMenu extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      placeholder: 'Wyszukaj...',
+    };
+    this.changeSearchInput = this.changeSearchInput.bind(this);
+    this.changeMenuLayout = this.changeMenuLayout.bind(this);
+    this.hiddenPlaceholder = this.hiddenPlaceholder.bind(this);
+    this.onSearchBlur = this.onSearchBlur.bind(this);
+  }
+
+  showSettings(event) {
+    event.preventDefault();
+  }
+
+  changeSearchInput(text) {
+    this.props.checkInputValue(text);
+  }
+
+  changeMenuLayout() {
+    this.props.changeLayout();
+  }
+
+  hiddenPlaceholder() {
+    this.setState({placeholder: ''});
+  }
+
+  onSearchBlur() {
+    this.setState({placeholder: 'Wyszukaj...'});
+  }
+
+  render() {
+    return (
+      <div className="menu-component">
+        <div className="menu-content">
+          <div className="search-bar">
+            <SearchInput
+              placeholder={this.state.placeholder}
+              onChange={(event) => this.changeSearchInput(event)}
+              onClick={this.hiddenPlaceholder}
+              onBlur={this.onSearchBlur}
+            />
+          </div>
+          <div className="button">
+            <FlatButton onClick={(event) => this.changeMenuLayout(event)}>
```

Comment:
po co dodawać funkcję w której później wywołuje się tylko funkcja z propsów? Po co przekazujesz event? Nie używasz go.
+            <FlatButton onClick={() => this.props.changeLayout()}>